transformers4rec/torch/utils/data_utils.py

```diff
@@ -356,7 +356,14 @@ def __init__(
             global_size=global_size,
             global_rank=global_rank,
             drop_last=drop_last,
-        ).map(self._get_pad_fn(sparse_max))
+            transforms=transforms,
+        )
+        if max_sequence_length:
+            # Apply padding
+            output_schema = loader.output_schema
+            sparse_feats = [col.name for col in output_schema if Tags.LIST in col.tags]
```

Comment:
Wouldn't we use `col_schema.is_list` instead of `Tags.LIST` to identify sparse features?

Reply:
yeah, that's what I was using as a first implementation but then I hit an edge case with the 2-d pretrained embeddings. In fact, the `is_list` property is checking if a shape has more than one dimension to decide if the feature is a list or not. In the 2-d pretrained case, the feature has two dimensions (batch_size x emb_dim) but it is not a list, so we'll get `col_schema.is_list` as True but the feature is not sparse. The longterm solution is maybe to extend the `is_list` property in merlin-core to differentiate between pre-trained embeddings and other features. In the meantime, I just kept the same logic we had previously, which is setting sparse features from the Tags.LIST